webhook/icinga.go

```diff
@@ -148,6 +148,10 @@ func updateOrCreateService(icinga icinga2.Client,
 	// Check if this is a heartbeat service. Adjust serviceData
 	// accordingly
 	if val, ok := alert.Labels["heartbeat"]; ok {
+		if alert.Status == "resolved" {
```

Comment:
Golang promotes to do assertions and early returns as early as possible within a function. This is to prevent doing work not needed. This is the case hear. `serviceVars` and `serviceData` is generated but never used for this case.

Reply:
I'm a fan of not repeating myself though, and I'd have to repeat the heartbeat check to properly do this earlier than here

Reply:
I've refactored heartbeat "detection" in the latest version of the PR. It's now done before creating the serviceData object.